### ISSUE TITLE
[parser bug] NPE on a long case statement

### ISSUE THREAD

**Issue by ahorek** [jruby/jruby#6441]
**Environment Information**
jruby 9.3.0.0-SNAPSHOT (2.6.5) 2020-10-21 6798b88442 Java HotSpot(TM) 64-Bit Server VM 11.0.1+13-LTS on 11.0.1+13-LTS +jit [mswin32-x86_64]

master only, most probably related to `no_sourceposition` refactor cc @enebo 

example
```
def test(num)
  case num
  when 55
  when 64
  when 65
  when 66
  when 67
  when 68
  when 69
  when 80
  when 81
  when 82
  when 83
  when 84
  when 85
  when 86
  when 87
  when 88
  when 96
  when 97
  when 98
  when 99
  when 100
  when 101
  when 112
  when 120
  when 121
  when 128
  when 130
  when 144
  when 145
  when 146
  when 147
  when 148
  when 149
  when 150
  when 160
  when 176
  when 177
  when 178
  when 179
  when 180
  when 181
  when 182
  when 183
  when 184
  when 185
  when 190
  when 200
  when 201
  when 202
  when 250
  when 251
  when 260
  when 261
  when 280
  when 281
  when 300
  when 301
  when 302
  when 320
  when 350
  when 500
    return "Video"
  end
end

puts test(500)
```

**Expected Behavior**
=> Video

**Actual Behavior**
```
Unhandled Java exception: java.lang.NullPointerException
java.lang.NullPointerException: null
                     yyparse at org/jruby/parser/RubyParser.java:1642
                     yyparse at org/jruby/parser/RubyParser.java:1597
                       parse at org/jruby/parser/RubyParser.java:5764
                       parse at org/jruby/parser/Parser.java:133
                       parse at org/jruby/parser/Parser.java:109
          parseFileAndGetAST at org/jruby/Ruby.java:2573
  parseFileFromMainAndGetAST at org/jruby/Ruby.java:2566
           parseFileFromMain at org/jruby/Ruby.java:2550
               parseFromMain at org/jruby/Ruby.java:1031
                 runFromMain at org/jruby/Ruby.java:952
               doRunFromMain at org/jruby/Main.java:393
                 internalRun at org/jruby/Main.java:287
                         run at org/jruby/Main.java:234
                        main at org/jruby/Main.java:206
```

**Comment by djberg96** [comment]
Coincidence? https://github.com/djberg96/sys-cpu/issues/11

It doesn't even have to be a big case statement. I don't think it's handling `case` without an explicit argument properly. Even this little script bombs:

```
def stuff(arg = nil)
  case
  when 0
    puts "zero"
  when 1
    puts "one"
  else
    puts "oops"
  end
end

stuff
```

**Comment by ahorek** [comment]
yeah, I've found it when testing sys-cpu. But this is a different bug then https://github.com/jruby/jruby/issues/6440

actually the latest stable 9.2.13.0 is fine, it's a regression in master

**Comment by djberg96** [comment]
> actually the latest stable 9.2.13.0 is fine, it's a regression in master

Eh? That script bombs with 9.2.13.0.

**Comment by headius** [comment]
Good find, thanks for being on the ball!

How could this be blowing up on such a simple case and still passing language specs?

**Comment by headius** [comment]
@djberg96 The case you point out does seem to be broken in 9.2.13 but it's a bug in the IR interpreter:

```
$ rvm jruby-9.2.13.0 do ruby blah.rb
Unhandled Java exception: java.lang.NullPointerException
java.lang.NullPointerException: null
  interpretAndGetNewIPC at org/jruby/ir/instructions/BSwitchInstr.java:99
              interpret at org/jruby/ir/interpreter/StartupInterpreterEngine.java:83
              interpret at org/jruby/ir/interpreter/InterpreterEngine.java:80
```

It is a separate issue and should be fixed for 9.2.14. Could you open a new issue for that please?

The original case by @ahorek works correctly in 9.2.13:

```
$ rvm jruby-9.2.13.0 do ruby blah.rb
Video
```

And it also works before the no_sourceposition refactor:

```
$ git log --oneline -1
2d12159ade (HEAD -> master) Merge pull request #6431 from MariuszCwikla/bug_in_initPlatform

$ ruby -v blah.rb
jruby 9.3.0.0-SNAPSHOT (2.6.5) 2020-10-21 2d12159ade OpenJDK 64-Bit Server VM 11.0.4+11 on 11.0.4+11 +jit [darwin-x86_64]
Video
```

**Comment by headius** [comment]
@djberg96 Disregard filing a new issue... I see now that #6440 is exactly that.

**Comment by djberg96** [comment]
@headius Ok. I didn't know (or had forgotten) that a bare case was even legal. Not sure I've ever done it, not on purpose anyway. :)

**Comment by enebo** [comment]
Hmm.  Fun this was introduced when I add @{n} position support to the parser.  It looks like I messed up the yystates growth if check.

**Comment by enebo** [comment]
Err initializeStates in that code was just wrong...fix shortly.

**Comment by enebo** [comment]
Fixed in 9a41d116a6303475e05c8105a671b21145438b4d.  I also added a regression spec to our suite since this is very impl specific.

**Comment by headius** [comment]
I believe the hash for the fix is e2598cebdfb32e139f2ef92ca63432b8b7f1c392.
